website/project/views/register.py

```diff
@@ -301,6 +255,74 @@ def node_registration_embargo_disapprove(auth, node, token, **kwargs):
     status.push_status_message('Your disapproval has been accepted and the embargo has been cancelled.', kind='success', trust=False)
     return redirect(redirect_url)
 
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+def node_registration_approve(auth, node, token, **kwargs):
+    """Handles disapproval of registration retractions
+    :param auth: User wanting to disapprove retraction
+    :return: Redirect to registration or
+    :raises: HTTPError if invalid token or user is not admin
+    """
+
+    if not node.pending_retraction:
+        raise HTTPError(http.BAD_REQUEST, data={
+            'message_short': 'Invalid Token',
+            'message_long': 'This registration is not pending a retraction.'
+        })
+
+    try:
+        node.registration_approval.approve(auth.user, token)
+        node.registration_approval.save()
```

Comment:
This line should likely be moved to an `else` block of this `try` - unless you expect it to potentially raised one of the handled exceptions, of course.